mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -1736,11 +1715,13 @@ static func::FuncOp createVerifierFunc(ModuleOp &module, const KernelIF &kernel,
           b.create<memref::StoreOp>(loc, valExt, arg1, ivs);
         });
   } else {
-    testResult = b.create<memref::CastOp>(loc, mr5DUnkType, arg0);
+    testResult = makeNDMemRef(b, arg0, 5);
```

Comment:
I forget exactly why I needed this.  I think I had a test case at a slightly different point in the pipeline.

Reply:
Despite the name, the only ops that makeNDMemRef() makes are ExpandShape and CollapseShape.  It just adjusts the rank of the reference;  I believe my problem was that CastOp wasn't enough to make it work for my side cases.